airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -326,7 +326,12 @@ def post_set_task_instances_state(*, dag_id: str, session: Session = NEW_SESSION
             detail=f"Task instance not found for task {task_id!r} on execution_date {execution_date}"
         )
 
-    if run_id and not session.query(TI).get({'task_id': task_id, 'dag_id': dag_id, 'run_id': run_id}):
+    if run_id and (
+        session.query(TI)
+        .filter(TI.task_id == task_id, TI.dag_id == dag_id, TI.run_id == run_id)
+        .one_or_none()
+        is None
+    ):
```

Comment:
tests/api_connexion/endpoints/test_task_instance_endpoint.py::TestPostSetTaskInstanceState::test_should_assert_call_mocked_api_when_run_id: sqlalchemy.exc.InvalidRequestError: Incorrect number of values in identifier to formulate primary key for query.get(); primary key columns are 'task_instance.task_id','task_instance.dag_id','task_instance.run_id','task_instance.map_index'

```suggestion
if run_id and not session.query(TI).get({'task_id': task_id, 'dag_id': dag_id, 'run_id': run_id, 'map_index': -1}):    ):
```

Reply:
Need to pull to fix static checks if any...pulling